src/services/recim/team.ts

```diff
@@ -0,0 +1,89 @@
+import http from '../http';
+import { Match } from './match';
+import { Participant } from './participant';
+
+export type Team = {
+  ID: number;
+  Name: string;
+  Status: string;
+  Logo: string;
+  Match: Match[];
+  Participant: Participant[];
+  MatchHistory: TeamMatchHistory[];
+  TeamRecord: TeamRecord[];
+};
+
+type CreatedTeam = {
+  ID: number;
+  Name: string;
+  StatusID: number;
+  ActivityID: number;
+  Logo: string;
+};
+
+type CreatedTeamParticipant = {
+  ID: number;
+  TeamID: number;
+  ParticipantID: number;
+  SignDate: string;
+  RoleTypeID: number;
+};
+
+type TeamMatchHistory = {
+  MatchID: number;
+  Opponent: Team;
+  OwnScore: number;
+  OpposingScore: number;
+  Status: string;
+  MatchStatusID: number;
+  Time: string;
+};
+
+export type TeamRecord = {
+  ID: number;
+  Name: string;
+  Win: number;
+  Loss: number;
+  Tie: number;
+};
+
+type UploadTeam = {
+  Name: string;
+  ActivityID: number;
+  Logo: string;
+};
+
+type PatchTeam = {
+  Name: string;
+  StatusID: number;
+  Logo: string;
+};
+
+//Team Routes
+const createTeam = (username: string, newTeam: UploadTeam): Promise<CreatedTeam> =>
+  http.post(`recim/Teams?username=${username}`, newTeam);
+
+const getTeamByID = (ID: number): Promise<Team> => http.get(`recim/Teams/${ID}`);
+
+const addParticipantToTeam = async (
+  username: string,
+  TeamID: number,
+  RoleID: number,
+): Promise<CreatedTeamParticipant> => {
+  var query = RoleID === null ? '' : `?RoleID=${RoleID}`;
+  return await http.post(`recim/Teams/Participants?username=${username}&TeamID=${TeamID}${query}`);
```

Comment:
RoleID and username should be part of the post body since they are object properties of a TeamParticipant. Also, you should be referencing the specific Teams resource in the URI path.
```suggestion
  return await http.post(`recim/Teams/${TeamID}/Participants`, {Username: username, RoleID: RoleID});
```

Reply:
This would require another viewmodel/class in the API and, for consistency sake, another type for the type {Username, RoleID} as I don't think it would be very clean to have an injected object like that. Right now I'll pass it as /Teams/teamID/username=username&roleID=roleID

Reply:
You should have a type for any data that you pass to the API, no matter how small of an object. (of course, if it is a native type like a string, bool, or int, it does not need its own type)

Data should not be sent in the query string of a post request.

Reply:
In that case, I'll make a new viewmodel for it on the API end and make that a new type.